static/js/log-search.js

```diff
@@ -24,12 +24,67 @@ let indexValues = [];
 
 $(document).ready(async () => {
     setSaveQueriesDialog();
+    let tpt = Cookies.get('queryMode');
+    let sessionQueryMode = sessionStorage.getItem('queryMode');
+
     let indexes = await getListIndices();
     if (indexes){
         originalIndexValues = indexes.map(item => item.index);
         indexValues = [...originalIndexValues];
     }
     initializeIndexAutocomplete();
+
+    // Function to set queryMode in both session storage and cookie
+    function setQueryMode(mode) {
+        Cookies.set('queryMode', mode, { expires: 7 }); // Set queryMode cookie with 7 days expiry
+        sessionStorage.setItem('queryMode', mode); // Set queryMode in session storage
+    }
+
+    // Function to remove queryMode cookie
+    function removeQueryModeCookie() {
+        Cookies.remove('queryMode');
+    }
+
+    // Function to check and manage queryMode cookie based on URL
+    function manageQueryModeCookie() {
+        const currentUrl = window.location.pathname;
+        const searchParams = new URLSearchParams(window.location.search);
+
+        // Check if the URL has the 'filterTab' parameter
+        const hasFilterTab = searchParams.has('filterTab');
+
+        // Check if current URL path ends with index.html or is /
+        if ((currentUrl.endsWith("index.html") || currentUrl === "/") && !hasFilterTab) {
+            // Prioritize cookie value over session storage
+            let queryMode = Cookies.get('queryMode');
+            if (queryMode !== undefined) {
+                setQueryMode(queryMode);
+                console.log("QueryMode set from cookie: " + queryMode);
+            } else if (sessionQueryMode !== null) {
+                setQueryMode(sessionQueryMode);
+                console.log("QueryMode set from session storage: " + sessionQueryMode);
+            } else {
+                setQueryMode('Builder');
+                console.log("QueryMode set to default: Builder");
+            }
+        } else if (hasFilterTab) {
+            // Save the current queryMode in session storage before removing the cookie
+            let currentQueryMode = Cookies.get("queryMode");
+            if (currentQueryMode !== undefined) {
+                sessionStorage.setItem('queryMode', currentQueryMode);
+                console.log("Storing queryMode in session storage: " + currentQueryMode);
+            }
+            removeQueryModeCookie(); // Remove queryMode cookie for URLs with 'filterTab' parameter
+        }
+    }
+
+    // Call manageQueryModeCookie initially on document ready
+    manageQueryModeCookie();
+
+    // Call manageQueryModeCookie whenever the URL changes
+    $(window).on('hashchange', manageQueryModeCookie);
+    $(window).on('popstate', manageQueryModeCookie);
+
     let queryMode = Cookies.get('queryMode');
     if (queryMode !== undefined){
         if(queryMode === "Builder"){
```

Comment:
We do not need to store it in session or remove the cookie. We can simply check if the filter tab is present and avoid triggering the tab in that case.

Reply:
I tried doing that but in that case, I am not able to get the original mode back, that's why I used session storage ...in this code, nothing is changing or creating problems to other tabs or URLs. I have thoroughly tested it by navigating to different tabs and trying out possible edge cases.

Reply:
Also, nothing is being changed here (neither session nor cookies),its just that I am removing cookies and getting it back as it would be stored in session when we are back on the logs tab.

Reply:
Yes, I understand this works, but I think this can be done in a simpler way. 
Can you try 
```
    if (queryMode !== undefined){
        const searchParams = new URLSearchParams(window.location.search);

        // Check if the URL has the 'filterTab' parameter
        const hasFilterTab = searchParams.has('filterTab');

        if (!hasFilterTab){
           //If filter tab is not present then do trigger.
            if(queryMode === "Builder"){
                $('.custom-code-tab a:first').trigger('click');
            }else{
                $('.custom-code-tab a[href="#tabs-2"]').trigger('click');
            }
        }
        // Add active class to dropdown options based on the queryMode selected.
        if(queryMode === "Builder"){
            $(".query-mode-option").removeClass("active");
            $("#query-mode-options #mode-option-1").addClass("active");
            $('#query-mode-btn span').html("Builder");
        }else{
            $(".query-mode-option").removeClass("active");
            $("#query-mode-options #mode-option-2").addClass("active");
            $('#query-mode-btn span').html("Code");
        }
    }